academic_validation_framework/validators/base.py

```
@@ -0,0 +1,359 @@
+    def _extract_metadata(self, research_output: Any) -> Dict[str, Any]:
+        """Extract metadata from research output."""
+        if isinstance(research_output, dict):
+            return research_output.get("metadata", {})
+        elif hasattr(research_output, "metadata"):
+            return research_output.metadata or {}
+        else:
+            return {}
+    
+    async def cleanup(self) -> None:
+        """Cleanup validator resources."""
+        pass
```

Ruff (0.11.9)

`BaseValidator.cleanup` is an empty method in an abstract base class, but has no abstract decorator

(B027)